File: backend/app/schemas/remote_plugin.py
"""远程插件 Pydantic schemas。"""
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class RemotePluginCreate(BaseModel):
    source_url: str
    default_enabled: bool = False


class RemotePluginOut(BaseModel):
    id: int
    name: str
    display_name: str
    description: str
    usage: str | None = None
    author: str
    source_url: str
    version: str
    latest_version: str | None = None
    update_available: bool = False
    last_update_check_at: datetime | None = None
    last_update_check_error: str | None = None
    lint_warnings: list[str] = Field(default_factory=list)
    event_subscriptions: list[dict[str, Any]] = Field(default_factory=list)
    capabilities: dict[str, Any] = Field(default_factory=dict)
    permissions: list[str] = Field(default_factory=list)
    enabled: bool
    default_enabled: bool = False
    installed_at: datetime | None = None

    class Config:
        from_attributes = True


class RegistryPluginOut(BaseModel):
    name: str
    display_name: str
    description: str
    author: str
    source_url: str
    version: str
    installed: bool

File: frontend/src/types/remotePlugin.ts
import type { PluginCapabilities, PluginEventSubscription } from "@/types/pluginContract";

export interface RemotePlugin {
  id: number;
  name: string;
  display_name: string;
  description: string;
  usage?: string | null;
  author: string;
  source_url: string;
  version: string;
  latest_version?: string | null;
  update_available?: boolean;
  last_update_check_at?: string | null;
  last_update_check_error?: string | null;
  lint_warnings?: string[];
  event_subscriptions?: PluginEventSubscription[];
  capabilities?: PluginCapabilities;
  permissions?: string[];
  enabled: boolean;
  default_enabled: boolean;
  installed_at: string | null;
}

export interface InstallRequest {
  source_url: string;
  default_enabled?: boolean;
}

export interface AccountPluginAction {
  account_ids: number[];
}

export interface RemotePluginUpdateCheckResponse {
  total: number;
  checked: number;
  update_available: number;
  failed: number;
}
